sycl/unittests/thread_safety/ThreadUtils.h

```diff
@@ -0,0 +1,28 @@
+#include <thread>
+#include <vector>
+
+class ThreadPool {
+public:
+  void clear() { MThreadPool.clear(); }
+
+  template <typename Func, typename... Args>
+  void enqueueNTimes(std::size_t N, Func &&func, Args &&... args) {
+    for (std::size_t i = 0; i < N; ++i)
+      enqueue(std::forward<Func>(func), std::forward<Args>(args)...);
+  }
+
+  template <typename Func, typename... Args>
+  void enqueue(Func &&func, Args &&... args) {
+    MThreadPool.push_back(
+        std::thread(std::forward<Func>(func), std::forward<Args>(args)...));
+  }
+
+  void wait() {
+    for (auto &t : MThreadPool) {
+      t.join();
+    }
```

Comment:
Since you create a new thread for each `enqueue`, the API looks quite convoluted.
Why not just creating a 1-shot `ThreadPool`, remove `clear()`, call `wait()` from `~ThreadPool()` or even better remove `wait()` interface and moved its code inside the destructor and just trust RAII.

Reply:
Thanks for review! But I'm not sure about this comment tho. I would rather call `wait` explicitly than expect that threads execution will start after destructor call. IMO, it's just more obvious. Besides, I tried to create the class as simple and universal as possible. I don't see any reason to adapt it for only one case.

Reply:
The problem is that this class is not universal. It does not even implement the ThreadPool suggested by the name...
So at the end this is why I find the API convoluted and not "as simple [...] as possible" for what is provided.
In a thread pool you would not create threads for every new work item. You would cache and park them waiting for new work item to be submitted...

Reply:
I see, thanks for explanation!